File: xcel6695-cs50-2017-x-pset6/xcel6695-cs50-2017-x-pset6/caesar.py
import cs50
import sys

def main():
  if len(sys.argv) != 2:
    print("Try again")
    exit(1)
  key = int(sys.argv[1])
  result = []
  print("Plaintext: ", end = "")
  message = cs50.get_string()
  for symbol in message:
    if symbol.isalpha():
      result.append(caesar(symbol, key))
    else:
      result.append(symbol)
  
  print("Ciphertext: ", end = "")              
  print("".join(result))
  exit(0)

def caesar(char, key):
  if char.isupper():
    return chr(((ord(char) - 65 + key) % 26) + 65)
  else:
    return chr(((ord(char) - 97 + key) % 26) + 97)
        
if __name__ == "__main__":
    main()
  

File: xcel6695-cs50-2017-x-pset2/xcel6695-cs50-2017-x-caesar/xcel6695-cs50-2017-x-caesar/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>

int lower_case(char a) 
{
    if (a >= 97 && a <= 122) 
        return 1;
    else 
        return 0;
}

int main(int argc, string argv[]) 
{
    while (argc != 2) 
    {
        printf("Usage: ./caesar k\n");
        return 1;
    }
    string name;
    printf("plaintext: ");
    name = get_string();
    int key = atoi(argv[1]);
    //char tmp;
    
    int len = strlen(name);
    printf("ciphertext: ");
    for (int i = 0; i < len; i++) 
    {
        if (isalpha(name[i])) 
        { 
            if (lower_case(name[i])) 
            {
                printf("%c", ((((name[i] - 97) + key) % 26) + 97));
            }
            else 
            {
                printf("%c", ((((name[i] - 65) + key) % 26) + 65));
            }
        }
        else
        {
            printf("%c", name[i]);
        }
    }
    printf("\n");
    return 0;
}

